### ISSUE TITLE
CONTRIBUTING.md Missing LocalHost Build/Test Instructions

### ISSUE THREAD

**Issue by jwwicks** [dotnet/docs#11825]
**Issue description**
As a new contributor I would like for the contributing guide to include DocFX build instructions so I can test my changes locally before submitting a PR. Currently there are no guidelines for setting up DocFX so you can test. 

Installing via Chocolatey, building with supplied config and serving locally doesn't produce a site suitable for testing. 

If the practice is being discouraged as per #627, #7734 etal... then perhaps a description of running the OpenPublishing PS file for a report would suffice.

Might be my newbie status but running that script produces 2 errors about permissions same as #627.
{"message":"Error occurred when restoring dependency repo from URL 'https://github.com/Microsoft/templates.docs.msft' branch 'master' into folder '_themes': You have no access permission to repository 'https://github.com/Microsoft/templates.docs.msft'","source":"OPS.Microsoft.OpenPublishing.Build.Applications.InitializeCRRTool.exe","file":null,"line":null,"message_severity":"Error","message_severity_string":"Error","code":null,"date_time":"2019-04-13T21:30:15.5108111Z","log_item_type":"User","log_item_type_string":"User","category":null}
{"message":"Error occurred when restoring dependency repo from URL 'https://github.com/Microsoft/templates.docs.msft.pdf' branch 'master' into folder '_themes.pdf': You have no access permission to repository 'https://github.com/Microsoft/templates.docs.msft.pdf'","source":"OPS.Microsoft.OpenPublishing.Build.Applications.InitializeCRRTool.exe","file":null,"line":null,"message_severity":"Error","message_severity_string":"Error","code":null,"date_time":"2019-04-13T21:30:15.5238121Z","log_item_type":"User","log_item_type_string":"User","category":null}

Could a private project GitHub Page get around the multiple repos dependencies cited in #627 etc...?

**Comment by mairaw** [comment]
Yeah, the answer from the engineering team was given on issue #11619. The local build depends on a repo that the community doesn't have access. I'd suggest that you log an issue at https://github.com/microsoftdocs/feedback requesting this kind of functionality since this is beyond the scope of this repo. We've been requesting that sort of functionality, but it's good if they hear directly from users as well.

## To work around this issue 

The following article covers some of the tools that you could use to edit locally:
https://docs.microsoft.com/en-us/contribute/get-started-setup-tools

The docs authoring pack has a preview functionality that you could see how the page would look like on docs. See if that would help.

And then after you submit a PR here, you get a build report from OPS in a few minutes. 

So, in short, the best route is to do your changes and then submit a PR to get a build going.

**Comment by jwwicks** [comment]
Thanks @mairaw, just me not wanting to break the build :)

It's more about the links than anything else especially if adding a page. The preview doesn't work real well for relative files.

[ValueTypes]: (../language-reference/keywords/value-types-table.md)
